### PR TITLE
WIP: Unique ID

### PR DESCRIPTION
**Pull request by jackwener**
#### What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### What does this PR do?
Add Unique ID generator including uuid and snowflake...

#### Which issue(s)/PR(s) this PR relates to?
None
  
#### Checklist：
- [ ] Documentation affected （Please add the label if documentation needs to be modified.)
- [ ] Incompatible （If it is incompatible, please describe it and add corresponding label.）
- [ ] Need to cherry-pick （If need to cherry-pick to some branches, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes：
Please confirm whether to reflect in release notes and how to describe:

### CODE REVIEW

**Comment by jackwener** [comment]
@yixinglu PTAL. Since I am not proficient in C++, there are many grammatical problems now, please help review.

**Review by yixinglu — COMMENTED** [review]
Performance needs to be considered, this function may be used to generate VID

File: src/meta/processors/id/GetSegmentIdProcessor.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+class GetSegmentIdProcessor : public BaseProcessor<cpp2::GetSegmentIdResp> {
+ public:
+  static GetSegmentIdProcessor* instance(kvstore::KVStore* kvstore) {
+    return new GetSegmentIdProcessor(kvstore);
+  }
+
+  void process(const cpp2::GetSegmentIdReq& req);
+
+ private:
+  mutable std::mutex lock_;
```
Comment (CPWstatic) [inline src/meta/processors/id/GetSegmentIdProcessor.h:23]: This lock only take afact to current instance, please refer to the LockUtils.

File: src/meta/processors/id/GetWorkerIdProcessor.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+class GetWorkerIdProcessor : public BaseProcessor<cpp2::GetWorkerIdResp> {
+ public:
+  static GetWorkerIdProcessor* instance(kvstore::KVStore* kvstore) {
+    return new GetWorkerIdProcessor(kvstore);
+  }
+
+  void process(const cpp2::GetWorkerIdReq& req);
+
+ private:
+  mutable std::mutex lock_;
```
Comment (CPWstatic) [inline src/meta/processors/id/GetWorkerIdProcessor.h:24]: The same problem, ditto.

File: src/meta/processors/id/GetWorkerIdProcessor.cpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "meta/processors/id/GetWorkerIdProcessor.h"
+
+namespace nebula {
+namespace meta {
+
+void GetWorkerIdProcessor::process(const cpp2::GetWorkerIdReq& req) {
+  const string& mac_address = req.get_mac_address();
```
Comment (CPWstatic) [inline src/meta/processors/id/GetWorkerIdProcessor.cpp:13]: Don't use mac address.

File: src/common/id/SegmentId.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      cur_ = segmentStart_;
+    }
+
+    return cur_;
+  }
+
+ private:
+  // wip: async
+  void fetchSegment() {
+    int64_t newSegment = client_.getSegmentId();
```
[the remark this answers is not in the document]
Comment (CPWstatic) [reply src/common/id/SegmentId.h:40]: std::move(future).via(runner).then([this] (auto& seg) { this.seg_ = seg;});

**Pull request closed without merging**